js/manager-utils.js

```
@@ -174,18 +173,361 @@ class ToastManager {
+        });
+    }
+
+    /**
+     * Clean up modal resources
+     * @param {string} modalId - ID of modal to cleanup
+     */
+    cleanupModal(modalId) {
+        const modalData = this.activeModals.get(modalId);
+        if (modalData) {
+            // Remove from DOM
+            if (modalData.modalElement && modalData.modalElement.parentNode) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)